core/src/main/java/org/neo4j/cypher/export/CypherResultSubGraph.java

```diff
@@ -205,6 +205,13 @@ public boolean contains( Relationship relationship )
                 .collect(Collectors.toSet());
     }
 
+    @Override
+    public Iterable<IndexDefinition> getIndexes(RelationshipType type) {
+        return indexes.stream()
+                .filter(idx -> StreamSupport.stream(idx.getRelationshipTypes().spliterator(), false).anyMatch(lb -> lb.equals(type)))
```

Comment:
should probably do equals on the name() to be sure, because RelationshipType is an interface so there is no symmetric equals